File: String/3138.Minimum_Length_of_Anagram_Concatenation/C++/3138.Minimum_Length_of_Anagram_Concatenation.cpp
class Solution {
public:
    int minAnagramLength(string s) 
    {
        //=================================//
        // Input type:                     //
        //   - string (s string)           //
        // Return type:                    //
        //   - int (result minimun length) //
        //=================================//

        /*Initialize*/
        ///// Length of s string /////
        int len_s = s.size(); 

        /*Vector-counting based loop traversal*/
        for (int s_idx = 1; (s_idx <= len_s); (s_idx++)) //Whole
        {
            ///// Check if the current length is dividable or not /////
            if ((len_s % s_idx)) { continue; }

            ///// Check if it's the minimun concatenated-anagrams or not /////
            if (isConcateAnag(len_s, s, s_idx)) { return s_idx; }

        } //Whole

        return len_s;
    }


    bool isConcateAnag(const int & len_s, const string & s, const int & record_cnter)
    {
        //=============================================//
        // Input type:                                 //
        //   - const int ref. (length of s string)     //
        //   - const string ref. (s string)            //
        //   - const int ref. (record summary counter) //
        // Return type:                                //
        //   - bool (result flag)                      //
        //=============================================//

        /*Initialize*/
        ///// Record vector /////
        vector<int> record_vect(26, 0);

        ///// Record string /////
        string record_str = s.substr(0, record_cnter);


        /*Whole process, flow*/
        ///// Step 1: Looped-traversal for the fixed-window size
        for (const char & record_chr: record_str) { ((record_vect[(record_chr - 'a')])++); } //Keep updating, recording

        ///// Step 2: Looped-traversal for the remainings /////
        for (int record_idx = record_cnter; (record_idx < len_s); (record_idx += record_cnter)) //Remainings
        {
            vector<int> sub_vect(26, 0); //Sub vector

            record_str = s.substr(record_idx, record_cnter); //Keep updating, overwriting

            for (const char & record_chr: record_str) { ((sub_vect[(record_chr - 'a')])++); } //Keep updating, recording

            ///// Check if the current vector matched conditions or not /////
            if (record_vect != sub_vect) { return false; }

        } //Remainings

        return true;
    }
};

File: String/3138.Minimum_Length_of_Anagram_Concatenation/Python/3138.Minimum_Length_of_Anagram_Concatenation.py
class Solution(object):
    def minAnagramLength(self, s):
        """
        :type s: str
        :rtype: int
        """
        #=======================================#
        # Array-counting based traversal method #
        #=======================================#

        ############
        #Initialize
        ##### Length of s string #####
        len_s = len(s)

        #####################################
        #Array-counting based loop traversal
        for s_idx in range(1, (len_s + 1)):

            ##### Check if the current length is dividable or not #####
            if ((len_s % s_idx)):
                continue

            ##### Check if it's the minimun concatenated-anagrams or not #####
            if (self.isConcateAnag(len_s, s, s_idx)):
                return s_idx

        return len_s


    def isConcateAnag(self, len_s, s, record_cnter):
        """
        :type len_s: int
        :type s: str
        :type record_cnter: int
        :rtype: bool
        """

        ############
        #Initialize
        ##### Record array #####
        record_arry = [0 for _ in range(26)]

        ##### Record string #####
        record_str = s[: record_cnter]


        ####################
        #Whole process/flow
        ##### Step 1: Looped-traversal for the fixed-window size #####
        for record_chr in record_str:
            record_arry[(ord(record_chr) - 97)] += 1 #Keep updating/recording

        ##### Step 2: Looped-traversal for the remainings #####
        for s_idx in range(record_cnter, len_s, record_cnter):
            sub_arry = [0 for _ in range(26)] #Sub-array #####

            record_str = s[s_idx: (s_idx + record_cnter)] #Keep updating/overwriting

            for record_chr in record_str:
                sub_arry[(ord(record_chr) - 97)] += 1 #Keep updating/recording

            ##### Check if the current array matched conditions or not #####
            if (sub_arry != record_arry):
                return False

        return True
